src/PiWeb.Import.Sdk/Modules/ImportFormat/IFileSource.cs

```diff
@@ -0,0 +1,76 @@
+﻿#region copyright
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+/* Carl Zeiss Industrielle Messtechnik GmbH        */
+/* Softwaresystem PiWeb                            */
+/* (c) Carl Zeiss 2024                             */
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#endregion
+
+namespace Zeiss.PiWeb.Import.Sdk.Modules.ImportFormat;
+
+#region usings
+
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+
+#endregion
+
+/// <summary>
+/// Represents a file.
+/// </summary>
+public interface IFileSource
+{
+    #region properties
+
+    /// <summary>
+    /// Gets the complete file name (including the extension) of this <see cref="IFileSource"/>.
+    /// </summary>
+    public string Name { get; }
+
+    /// <summary>
+    /// Gets the file name (without extension) of this <see cref="IFileSource"/> or <c>null</c> if the file has no name.
+    /// </summary>
+    public string? BaseName { get; }
+
+    /// <summary>
+    /// Gets the file extension (including the period ".") of this <see cref="IFileSource"/>.
+    /// </summary>
+    /// <remarks>
+    /// This method obtains the extension of path by searching path for a period ("."), starting from the last character in
+    /// <see cref="Name"/> and continuing toward its first character. If a period is found before a directory separator, the returned
+    /// string contains the period and the characters after it; otherwise, <c>null</c> is returned.
+    /// </remarks>
+    public string? Extension { get; }
+
+    /// <summary>
+    /// Gets the full file name and path information (starting with the import folder) of this <see cref="IFileSource"/>.
+    /// </summary>
+    public string Path { get; }
```

Comment:
Lets call this RelativePath to distinguish it from other path related properties we might introduce in later versions of this API.